grepos/app/src/main/java/net/boondockradio/grepos/MainActivity.java

```diff
@@ -22,51 +23,78 @@
 
 public class MainActivity extends AppCompatActivity {
 
-    private final String TAG = "grepos";
+    private final String TAG = MainActivity.class.getSimpleName();
+    private final int LOAD_MORE_THRESHOLD = 1;
+    private final String PER_PAGE = "50";
 
     private List<Repository> mRepositoryItems;
-    private RecyclerView mRecyclerView;
-    private RecyclerView.Adapter mAdapter;
+    private RepositoryAdapter mAdapter;
+    private ProgressBar mProgress;
+
+    private int mPage = 0;
+
+    private int total = 0;
+    private int last = 0;
+    private boolean isLoading = false;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        mProgress = (ProgressBar) findViewById(R.id.progress_main_activity);
+        mRepositoryItems = new ArrayList<>();
+
+        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recycler_main_activity);
+        recyclerView.setHasFixedSize(true);
+
+        final LinearLayoutManager layoutManager = new LinearLayoutManager(this);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+
+        recyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+                total = layoutManager.getItemCount();
+                last = layoutManager.findLastVisibleItemPosition();
+
+                if (dy < 0) {
+                    return;
```

Comment:
i want to return if dy is negative but this code is wrong. should be move head in this function.